services/controlplane/client/client_test.go

```diff
@@ -0,0 +1,329 @@
+			identity: &identity.Namespace{
+				Namespace:    "valid-namespace",
+				HostedSecret: validSecret,
+			},
+			revisionID: "2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+
+			responseBody: string(body),
+
+			wantDestination: expectedDestination,
+			wantPath:        "/workspaces/destinationHistory/2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+			wantErr:         nil,
+		},
+		{
+			name: "valid request in namespace with region",
+			identity: &identity.Namespace{
+				Namespace:    "valid-namespace",
+				HostedSecret: validSecret,
+			},
+			revisionID: "2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+			region:     "eu",
+
+			responseBody: string(body),
+
+			wantDestination: expectedDestination,
+			wantPath:        "/workspaces/destinationHistory/2ENkYQ1MR9f83YhqS5k7uTSe5XH?region=eu",
+			wantErr:         nil,
+		},
+		{
+			name: "invalid JSON",
+			identity: &identity.Workspace{
+				WorkspaceID:    "valid-workspace-id",
+				WorkspaceToken: validSecret,
+			},
+			revisionID: "2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+
+			responseBody: `<html>Hello</html>`,
+
+			wantDestination: backendconfig.DestinationT{},
+			wantPath:        "/workspaces/destinationHistory/2ENkYQ1MR9f83YhqS5k7uTSe5XH",
+			wantErr:         fmt.Errorf("unmarshal response body: invalid character '<' looking for beginning of value"),
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.responseStatus == 0 {
+				tc.responseStatus = http.StatusOK
+			}
+
+			s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				u, _, _ := r.BasicAuth()
+
+				require.Equal(t, "application/json", r.Header.Get("Content-Type"))
+				require.Contains(t, r.UserAgent(), "control-plane/features")
+				require.Equal(t, validSecret, u)
+				require.Equal(t, http.MethodGet, r.Method)
+				require.Equal(t, tc.wantPath, r.URL.String())
+
+				w.WriteHeader(tc.responseStatus)
+				fmt.Fprint(w, tc.responseBody)
+			}))
+			defer s.Close()
+
+			c := client.NewClient(
+				s.URL,
+				tc.identity,
+				client.WithHTTPClient(s.Client()),
+				client.WithRegion(tc.region),
+			)
+
+			destination, err := c.DestinationHistory(context.Background(), tc.revisionID)
+			if tc.wantErr != nil {
+				require.EqualError(t, err, tc.wantErr.Error())
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tc.wantDestination, destination)
+			}
+		})
+	}
+}
+
+func TestRetriesTimeout(t *testing.T) {
+	t.Log("all methods should exhibit the same retry and timeout behavior")
+	methods := []struct {
+		fn   func(*client.Client) error
+		name string
+	}{
+		{
+			name: "SendFeatures",
+			fn: func(c *client.Client) error {
+				return c.SendFeatures(context.Background(), "test", []string{"feature1", "feature2"})
+			},
+		},
+		{
+			name: "DestinationHistory",
+			fn: func(c *client.Client) error {
+				_, err := c.DestinationHistory(context.Background(), "test")
+				return err
+			},
+		},
+	}
+
+	for _, m := range methods {
+		t.Run(m.name, func(t *testing.T) {
```

Comment:
We would need to capture the [range variable](https://go.dev/blog/subtests#:~:text=capture%20range%20variable) here.